collections/ansible_collections/cloudkit/service/plugins/filter/find_template_roles.py

```diff
@@ -162,11 +165,17 @@ class NodeSet(Base):
     size: int
 
 
+class TemplateTypeEnum(Enum):
+    cluster = "cluster"
+    vm = "vm"
+
+
 class Metadata(Base):
     """Metadata about the template"""
 
     title: str
     description: str | None = None
+    template_type: str | None = None  # "cluster" or "vm"
     default_node_request: list[NodeRequest]
     allowed_resource_classes: list[str] | None = None
```

Comment:
These two parameters don't make sense in VM context, I think we would need a metadata class for clusters, and one for vms